yt/visualization/volume_rendering/off_axis_projection.py

```diff
@@ -380,32 +383,90 @@ def temp_weightfield(field, data):
         fields.append(vol.weight_field)
 
     mylog.debug("Casting rays")
+    index = data_source.ds.index
+    lens = camera.lens
+    if isinstance(index, OctreeIndex) and isinstance(lens, PlaneParallelLens):
+        fields.extend(("index", k) for k in "xyz")
+        fields.append(("index", "dx"))
+
+        data_source.get_data(fields)
+        # We need the width of the plot window in projected coordinates,
+        # i.e. we ignore the z-component
+        wmax = width[:2].max()
+
+        # Normalize the positions & dx so that they are in the range [-0.5, 0.5]
+        xyz = np.stack(
+            [
+                ((data_source["index", k] - center[i]) / wmax).to("1").d
+                for i, k in enumerate("xyz")
+            ],
+            axis=-1,
+        )
+
+        for idim, periodic in enumerate(data_source.ds.periodicity):
+            if not periodic:
+                continue
+            # Wrap into [-0.5, +0.5]
+            xyz[..., idim] = (xyz[..., idim] + 0.5) % 1 - 0.5
+
+        dx = (data_source["index", "dx"] / wmax).to("1").d
```

Comment:
are octree grids always cubes? I naively expected to see some handling of `dy` and `dz` as well, but this all seems to assume dx=dy=dz.

Reply:
I have added a comment in that sense. The octree grids are always cubic in the astronomical codes I know, but it might not be the general case. I've added a comment in that direction and an explicit check.